src/parser/shared/modules/items/bfa/raids/battleofdazaralor/WardOfEnvelopment.js

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS/index';
+import ITEMS from 'common/ITEMS/index';
+import ItemLink from 'common/ItemLink';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import ItemHealingDone from 'interface/others/ItemHealingDone';
+import Abilities from 'parser/core/modules/Abilities';
+
+const MAX_ALLIES_HIT = 5;
+const ACTIVATION_COOLDOWN = 120; // seconds
+
+/**
+ * Use: Envelop up to 5 allies in the targeted area in shared shields for 10 sec, preventing up to [x] damage taken by any of them. Damage prevented is increased when affecting multiple allies. (2 Min Cooldown)
+ * 
+ * Allies affected have the Enveloping Protection buff, lasts 10 seconds or until all damage is absorbed.
+ * Hitting 1 ally gives the base value, each additional ally hit adds 7.5% of the base value. Maximum of 5 people hit for 30% extra on top of the base value.
+ */
+class WardOfEnvelopment extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+  }
+
+  absorbUsed = 0;
+  absorbWasted = 0;
+  targetsHit = 0;
+  firstExpiration = false;
+  uses = 0;
+  
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrinket(ITEMS.WARD_OF_ENVELOPMENT.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.ENVELOPING_PROTECTION]), this.onCast);
```

Comment:
`spell()` method takes either a SPELL object or array of SPELL objects. Since you have only 1 object in the argument, it's better to just use 
```suggestion
    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.ENVELOPING_PROTECTION), this.onCast);
```
Same thing in other event listeners.